src/basic/mountpoint-util.c

```diff
@@ -509,3 +512,19 @@ int mount_propagation_flags_from_string(const char *name, unsigned long *ret) {
                 return -EINVAL;
         return 0;
 }
+
+int make_mount_point_inode(const char *source, const char *dest, mode_t mode)
+{
+        struct stat st;
+
+        assert(source);
+        assert(dest);
+
+        if (stat(source, &st) < 0)
+                return -errno;
+
+        if (S_ISDIR(st.st_mode))
+                return mkdir_p_label(dest, mode);
+        else
+                return touch_file(dest, true, USEC_INFINITY, UID_INVALID, GID_INVALID, mode & ~0111);
+}
```

Comment:
maybe add two flavours for this, one that takes a source path and one that takes a mode_t. And the former is a wrapper around the latter, and just calls stat() first. After all in many cases we'll already have issued stat() first.

Reply:
Thought about that, but the only place where we had a stat ready was in bus_machine_method_bind_mounts which is a runtime one-op, so shouldn't be in anyone's critical path, so kept it simpler. But it might be useful in the future, so will rework as suggested.

Reply:
Passing mode_t felt awkward, instead choose a stat pointer which looks nicer to me.